tests/kernel/update_kernel.pm

```diff
@@ -0,0 +1,209 @@
+# SUSE's openQA tests
+#
+# Copyright © 2017 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+# Summary: This module installs maint update under test for kernel/kgraft to ltp work image
+# Maintainer: Ondřej Súkup osukup@suse.cz
+
+use 5.018;
+use warnings;
+use strict;
+use base 'opensusebasetest';
+use testapi;
+use utils;
+use qam;
+
+sub update_kernel {
+    my ($repo, $patch) = @_;
+
+    fully_patch_system;
+
+    zypper_call("ar $repo kernel-update");
+    zypper_call("ref");
+
+    if (!is_patch_needed($patch)) {
+        zypper_call("in -l -t patch $patch", exitcode => [0, 102, 103], log => 'zypper.log');
+    }
+    else {
+        die "Patch isn't needed";
+    }
+}
+
+sub mod_rpm_info {
+    my $module = shift;
+
+    script_run("rpm -qf /$module");
+    save_screenshot;
+    script_run("modinfo /$module");
```

Comment:
I recommend in general to use either `assert_script_run` or `script_run` with explicit check on the return value.